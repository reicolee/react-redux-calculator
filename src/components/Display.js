import React from 'react';
import PropTypes from 'prop-types';
import Box from '@material-ui/core/Box';
import { Grid } from '@material-ui/core';

function Display({ displayValue }) {
  return (
    <Grid container item xs={9} justify="flex-end">
      {displayValue}
    </Grid>
  );
}

Display.propTypes = {
  displayValue: PropTypes.string.isRequired
};

export default Display;
