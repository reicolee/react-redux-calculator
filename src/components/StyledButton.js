import React from 'react';
import Button from '@material-ui/core/Button';

function StyledButton({ children, handleClick, type }) {
  const color =
    type === 'operator'
      ? 'secondary'
      : type === 'digit'
      ? 'primary'
      : 'default';

  return (
    <>
      <Button
        variant="outlined"
        color={color}
        size="medium"
        onClick={() => handleClick(children)}
        styled={{ borderRadious: 'none' }}
      >
        {children}
      </Button>
    </>
  );
}

export default StyledButton;
