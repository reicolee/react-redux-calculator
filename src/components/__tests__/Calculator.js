import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import { shallow, mount } from 'enzyme';

import Calculator from '../Calculator';
import Display from '../Display';
import StyledButton from '../StyledButton';

import rootReducer from '../../reducers';
import renderer from 'react-test-renderer';

describe('<Calculator /> renders properly', () => {
  let store;
  let elements;
  let wrapper;

  beforeAll(() => {
    store = createStore(rootReducer, applyMiddleware(thunk));
    elements = (
      <Provider store={store}>
        <Calculator />
      </Provider>
    );
    wrapper = mount(elements);
  });

  it('match snapshot', () => {
    const component = renderer.create(elements);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a Display Component', () => {
    expect(wrapper.find(Display).length).toBe(1);
  });

  it('renders 20 StyledButton Components', () => {
    expect(wrapper.find(StyledButton).length).toBe(20);
  });
});

describe('Test StyledButton Component', () => {
  it('match snapshot', () => {
    const handleClick = jest.fn();
    const component = renderer.create(
      <StyledButton handleClick={handleClick}>7</StyledButton>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('simulates click events', () => {
    const handleClick = jest.fn();
    const button = mount(
      <StyledButton handleClick={handleClick}>7</StyledButton>
    );

    button.find('button').simulate('click');
    expect(handleClick).toHaveBeenCalled();
  });
});
