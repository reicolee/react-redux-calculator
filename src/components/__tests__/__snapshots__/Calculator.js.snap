// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calculator /> renders properly match snapshot 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthXs"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-flex-end MuiGrid-grid-xs-9"
      >
        0
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          AC
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          MS
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          MR
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          /
        </span>
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          7
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          8
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          9
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          x
        </span>
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          4
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          5
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          6
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          -
        </span>
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          1
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          2
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          3
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          +
        </span>
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          0
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          .
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          ±
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          =
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`<Calculator /> renders properly renders correctly 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthXs"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-flex-end MuiGrid-grid-xs-9"
      >
        0
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          AC
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          MS
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          MR
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          /
        </span>
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          7
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          8
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          9
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          x
        </span>
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          4
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          5
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          6
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          -
        </span>
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          1
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          2
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          3
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          +
        </span>
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          0
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          .
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          ±
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        styled={
          Object {
            "borderRadious": "none",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          =
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Test StyledButton Component match snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-outlined"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  styled={
    Object {
      "borderRadious": "none",
    }
  }
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    7
  </span>
</button>
`;
